### PR TITLE
chore(ipfs-daemon,ipfs-topology): Use logger in IpfsDaemon and IpfsTopology

### PR DESCRIPTION
**Pull request by stbrody**
builds on https://github.com/ceramicnetwork/js-ceramic/pull/1021

### CODE REVIEW

**Comment by stbrody** [comment]
I'm really not sure what's going on with the test failure.  I seem to recall the multiformats library not playing nicely with jest, and I guess by moving the `buildIpfsConnection` call from `ceramic-cli-utils.ts` to `ceramic-daemon.ts` now it's getting picked up by the test file where it previously was not.  Not really sure what to do about it though.  I tested running the daemon from the CLI and everything works fine, it's only an issue in the test.

I could use some assistance here @oed @ukstv as I'm basically stuck on where to go from here

**Comment by stbrody** [comment]
Changing 
```
// @ts-ignore
import multiformats from 'multiformats/basics'
// @ts-ignore
import legacy from 'multiformats/legacy'
```
to
```
import multiformats from 'multiformats/cjs/src/basics-import.js'
import legacy from 'multiformats/cjs/src/legacy.js'
```

in `build-ipfs-connection.util.ts` and `ipfs-daemon.ts` makes all the tests pass, but then starting the daemon on the CLI fails with:
```
> ceramic daemon
internal/modules/cjs/loader.js:430
      throw e;
      ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './cjs/src/basics-import.js' is not defined by "exports" in /home/spencer/js-ceramic3/node_modules/multiformats/package.json
    at throwExportsNotFound (internal/modules/esm/resolve.js:285:9)
    at packageExportsResolve (internal/modules/esm/resolve.js:508:3)
    at resolveExports (internal/modules/cjs/loader.js:424:36)
    at Function.Module._findPath (internal/modules/cjs/loader.js:464:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:802:27)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/spencer/js-ceramic3/packages/ipfs-daemon/lib/ipfs-daemon.js:14:44)
    at Module._compile (internal/modules/cjs/loader.js:999:30) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

:cry:

**Comment by stbrody** [comment]
The `moduleNameMapper` trick seems to have fixed things.  Thanks for the tip @ukstv!  This is now ready for review

File: packages/cli/src/ceramic-daemon.ts [hunk 1]
```diff
@@ -185,10 +184,11 @@ class CeramicDaemon {
    * @param opts - Ceramic daemon options
    */
   static async create (opts: CreateOpts): Promise<CeramicDaemon> {
-    const { ipfs } = opts
-
     const ceramicConfig = makeCeramicConfig(opts)
 
+    const ipfs = await buildIpfsConnection(
```
[the remark this answers is not in the document]
Comment (ukstv) [reply packages/cli/src/ceramic-daemon.ts:189]: Whoops, true.

**Review by ukstv — APPROVED** [review]

**Comment by ukstv** [comment]
Heck, I miss Scala `implicit` context. Would make life easier here.

**Pull request merged**